pychromecast/dial.py

```diff
@@ -177,22 +177,33 @@ def get_device_info(  # pylint: disable=too-many-locals
     try:
         if services is None:
             services = [ServiceInfo(SERVICE_TYPE_HOST, (host, 8009))]
-        _, status = _get_status(
-            services,
-            zconf,
-            "/setup/eureka_info?params=device_info,name",
-            True,
-            timeout,
-            context,
-        )
+
+        try:
+            _, status = _get_status(
+                services,
+                zconf,
+                "/setup/eureka_info?params=device_info,name",
+                True,
+                timeout / 2,
+                context,
+            )
+        except (urllib.error.HTTPError, urllib.error.URLError):
+            _, status = _get_status(
+                services,
+                zconf,
+                "/setup/eureka_info?params=device_info,name",
+                False,
+                timeout / 2,
+                context,
+            )
 
         cast_type = CAST_TYPE_CHROMECAST
         display_supported = True
         friendly_name = status.get("name", "Unknown Chromecast")
         manufacturer = "Unknown manufacturer"
         model_name = "Unknown model name"
         multizone_supported = False
-        udn = None
+        udn = status.get("ssdp_udn", None)
```

Comment:
What's up with this change?
What do you get back from your device when fetching `/setup/eureka_info?params=device_info,name"` ?

Reply:
Response is different for insecure endpoint. This ensures that udn is correct in both cases.

Insecure endpoint:
```
{
  "bssid": "",
  "build_version": "374976",
  "cast_build_revision": "1.68.374976",
  "connected": true,
  "ethernet_connected": true,
  "has_update": false,
  "hotspot_bssid": "bssid",
  "ip_address": "ip",
  "locale": "en-US",
  "location": {
    "country_code": "US",
    "latitude": 255,
    "longitude": 255
  },
  "mac_address": "00:00:00:00:00:00",
  "name": "SHIELD",
  "opt_in": {
    "crash": false,
    "opencast": false,
    "stats": false
  },
  "public_key": "key",
  "release_track": "",
  "setup_state": 60,
  "setup_stats": {
    "historically_succeeded": true,
    "num_check_connectivity": 0,
    "num_connect_wifi": 0,
    "num_connected_wifi_not_saved": 0,
    "num_initial_eureka_info": 0,
    "num_obtain_ip": 0
  },
  "ssdp_udn": "udn",
  "ssid": "",
  "time_format": 1,
  "tos_accepted": true,
  "uptime": 5335.944937,
  "version": 12,
  "wpa_configured": false,
  "wpa_state": 0
}
```

Secure endpoint:
```
{
  "device_info": {
    "4k_blocked": 0,
    "capabilities": {
      "aogh_supported": true,
      "assistant_supported": true,
      "audio_hdr_supported": false,
      "audio_surround_mode_supported": false,
      "ble_supported": true,
      "bluetooth_audio_sink_supported": true,
      "bluetooth_audio_source_supported": true,
      "bluetooth_supported": true,
      "cloud_groups_supported": true,
      "cloudcast_supported": true,
      "content_filters_supported": true,
      "disable_google_dns_supported": true,
      "display_supported": false,
      "fdr_supported": false,
      "hdmi_prefer_50hz_supported": false,
      "hdmi_prefer_high_fps_supported": false,
      "hotspot_supported": true,
      "https_setup_supported": true,
      "input_management_supported": true,
      "keep_hotspot_until_connected_supported": true,
      "multi_user_supported": true,
      "multichannel_group_supported": true,
      "multizone_supported": true,
      "night_mode_supported": true,
      "night_mode_supported_v2": true,
      "opencast_supported": false,
      "preview_channel_supported": true,
      "reboot_supported": true,
      "remote_ducking_supported": true,
      "renaming_supported": true,
      "set_network_supported": true,
      "setup_supported": true,
      "sleep_mode_supported": true,
      "stats_supported": true,
      "system_sound_effects_supported": false,
      "ui_flipping_supported": true,
      "user_eq_supported": true,
      "wifi_auto_save_supported": true,
      "wifi_supported": true
    },
    "cloud_device_id": "id",
    "factory_country_code": "DE",
    "hotspot_bssid": "bssid",
    "local_authorization_token_hash": "hash",
    "mac_address": "mac",
    "manufacturer": "Google Inc.",
    "model_name": "Google Home Mini",
    "product_name": "mushroom",
    "public_key": "key",
    "ssdp_udn": "udn",
    "uma_client_id": "id",
    "uptime": 59169.998497,
    "weave_device_id": ""
  },
  "name": "Google Home"
}
```

Reply:
What happens if you instead fetch `/setup/eureka_info?options=detail` on the Nvidia Shield device?

Reply:
I get the same output.

Reply:
OK, I see.

Instead of doing `udn = status.get("ssdp_udn", None)` can you please move it to an else clause:
```py
if "device_info" in status:
    ...
else:
    udn = status.get("ssdp_udn", None)
```

Reply:
Done